src/compiler/core.ts

```diff
@@ -2417,7 +2417,13 @@ namespace ts {
      * Takes a string like "jquery-min.4.2.3" and returns "jquery"
      */
     export function removeMinAndVersionNumbers(fileName: string) {
-        return fileName.replace(/((?:\.|-)min(?=\.|$))|((?:-|\.)\d+)/g, "");
+        const match = /((\w|(-(?!min)))+)(\.|-)?.*/.exec(fileName);

```

Comment:
Doesn't this also take something like "foo.bar.js", and turn it into just "foo.js"? Is that intended?

Reply:
I just whipped up the below, which seems to work. Take a look and see what you think.

Note: I don't think we need to worry about .jsx files being library files, or trying to handle suffixes like "-beta" after the min or version.

```javascript
function removeMinAndVersionNumbers(path) {
  // Match a "." or "-" followed by a version number or 'min', that occurs right before the final ".js"
  let trailingMinOrVersion = /[.-]((min)|(\d+(\.\d+)*))\.js$/;

  // The "min" or version may both be present, in either order, so try applying the above twice.
  // Note the final ".js" is part of the match, so add this back as the substitution.
  let result = path.replace(trailingMinOrVersion, ".js").replace(trailingMinOrVersion, ".js");
  return result;
}
```

Reply:
This doesn't handle `jquery.6.min.js`; not sure how common that is

Reply:
Yes it does. I just tried it in the Console.
```
removeMinAndVersionNumbers("jquery.6.min.js");
"jquery.js"
```